configure.ac

```diff
@@ -136,43 +136,58 @@ AC_CHECK_HEADER(valgrind/memcheck.h, [],
 fi
 
 AC_CACHE_CHECK(whether ld accepts --version-script, ac_cv_version_script,
-    if test -n "`$LD --help < /dev/null 2>/dev/null | grep version-script`"; then
+    [if test -n "`$LD --help < /dev/null 2>/dev/null | grep version-script`"; then
         ac_cv_version_script=yes
     else
         ac_cv_version_script=no
-    fi)
+    fi])
 
 AM_CONDITIONAL(HAVE_LD_VERSION_SCRIPT, test "$ac_cv_version_script" = "yes")
 
 dnl Disable symbol versioning with icc + ipo, with it ipo is disabled by icc.
 dnl The gcc equivalent ipo (-fwhole-program) seems to work fine.
 AS_IF([case "$CFLAGS" in
```

Comment:
I wanted to reign it in for this commit, but def. for another PR. Also: cache the compiler checks, AC_DEFINE HAVE_ATOMIC, re-arrange all the compiler checks next to each other, consistent indenting (I try to copy what already exists but it is all over the place and however I indent it looks ugly).